api/src/main/java/org/pmiops/workbench/rdr/RdrExportServiceImpl.java

```diff
@@ -194,26 +195,31 @@ private RdrResearcher toRdrResearcher(DbUser dbUser) {
                     RdrExportEnums.ethnicityToRdrEthnicity(
                         dbDemographicSurvey.getEthnicityEnum()))));
       }
-      if (dbDemographicSurvey.getSexAtBirthEnum() != null) {
-        researcher.setSexAtBirth(
-            dbDemographicSurvey.getSexAtBirthEnum().stream()
-                .map(RdrExportEnums::sexAtBirthToRdrSexAtBirth)
-                .collect(Collectors.toList()));
-      }
+      researcher.setSexAtBirth(
+          Optional.ofNullable(
+                  dbDemographicSurvey.getSexAtBirthEnum().stream()
+                      .map(RdrExportEnums::sexAtBirthToRdrSexAtBirth)
+                      .collect(Collectors.toList()))
+              .orElse(new ArrayList<SexAtBirth>()));
+      researcher.setGender(
+          Optional.ofNullable(
+                  dbDemographicSurvey.getGenderIdentityEnumList().stream()
+                      .map(RdrExportEnums::genderToRdrGender)
+                      .collect(Collectors.toList()))
+              .orElse(new ArrayList<Gender>()));
       researcher.setDisability(
           RdrExportEnums.disabilityToRdrDisability(dbDemographicSurvey.getDisabilityEnum()));
-    }
 
-    if (null != dbDemographicSurvey && dbDemographicSurvey.getRaceEnum() != null) {
       researcher.setRace(
-          dbDemographicSurvey.getRaceEnum().stream()
-              .map(RdrExportEnums::raceToRdrRace)
-              .collect(Collectors.toList()));
-    } else {
-      researcher.setRace(new ArrayList<>());
+          Optional.ofNullable(
+                  dbDemographicSurvey.getRaceEnum().stream()
+                      .map(RdrExportEnums::raceToRdrRace)
+                      .collect(Collectors.toList()))
+              .orElse(new ArrayList<Race>()));
```

Comment:
Do you want the race field to be written to, or should the set of entries be fixed?

Clients might do `researcher.getRace().add(Race.DAYTONA_500);`, which works, but is counterintuitive.

Reply:
no RDR needs to simple take whatever value we give to them and not manipulate them